src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -370,10 +378,11 @@
       (if puppetdb-query
         req
         (let [query-map (create-query-map req param-spec parse-fn)
+              query-uuid (str (java.util.UUID/randomUUID))
               pretty-print (:pretty query-map
                                     (get-in req [:globals :pretty-print]))]
           (-> req
-              (assoc :puppetdb-query query-map)
+              (assoc :puppetdb-query query-map :puppetdb-query-uuid query-uuid)
               (assoc-in [:globals :pretty-print] pretty-print))))))))
```

Comment:
I just pushed a change to this part, just simplifying it by adding the query-uuid in the existing assoc call